src/pup/plugins/pip_install.py

```diff
@@ -25,13 +26,33 @@ def usable_in(ctx):
         )
 
     def __call__(self, ctx, dsp):
+        python_path = str(ctx.python_runtime_dir / ctx.python_rel_exe)
+
+        platform_flags = []
+        if ctx.pip_platform:
+            platform_flags.append('--platform={}'.format(ctx.pip_platform))
+            platform_flags.append('--only-binary=:all:')
+            # TODO: This should probably be done with the spawn helpers, but
+            #       done this way as a proof of concept
+            site_packages_path = subprocess.check_output([
+                python_path,
+                '-c',
+                'import site; print(site.getsitepackages()[0], end="")'
+            ])
+            site_packages_path = site_packages_path.decode("utf-8")
+            if os.path.isdir(site_packages_path):
+                platform_flags.append('--target')
```

Comment:
In which cases is this needed? AFAICT, `pip` is running within the just created Python environment (not virtual, an actual and complete+relocatable install) and will install to the "proper" `site-packages`... What am I missing?

Reply:
We want to use `--platform` flag to be able to specify a platform older than the current running system. Specially, for the macOS builds, we might be running this in macOS 12, but want wheels compatible with 10.15, etc.
Unfortunately we cannot use the `--platform` flag without adding the `--target` flag as well.